### PR TITLE
Follow all group member

### PR DESCRIPTION
**Pull request by habibillah**
this is a replacement for #82 
new endopoint: `/api/follow/group/{id}`

### CODE REVIEW

**Comment by igaponov** [comment]
I suggest to use \Liip\FunctionalTestBundle\Test\WebTestCase instead of yours, so you'll not have problems with cleaning your data after tests. It's already configured for test environment, just move all data to doctrine fixtures.

**Comment by habibillah** [comment]
Noted

File: backend/src/Civix/ApiBundle/Tests/Controller/FollowControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    {
+
+        $client = static::createClient();
+        $client->setServerParameter("HTTP_Token", $this->followerToken);
+
+        $client->request('POST', "/api/follow/group/{$this->group1->getId()}");
+
+        $response = $client->getResponse();
+
+        $this->assertSame(404, $response->getStatusCode());
```
Comment (austinpapp) [inline backend/src/Civix/ApiBundle/Tests/Controller/FollowControllerTest.php]: thank you for testing failures.

File: backend/src/Civix/CoreBundle/Entity/Group.php [hunk 2]
```diff
@@ -500,6 +512,7 @@ public function init()
         $this->groupType = self::GROUP_TYPE_COMMON;
         $this->membershipControl = self::GROUP_MEMBERSHIP_PUBLIC;
         $this->petitionPerMonth = self::COUNT_PETITION_PER_MONTH;
+        $this->transparency = self::GROUP_TRANSPARENCY_PUBLIC;
```
Comment (austinpapp) [inline backend/src/Civix/CoreBundle/Entity/Group.php]: we are always defaulting to public, correct?

**Comment by austinpapp** [comment]
@habibillah what is the test `WebTestCase` testing?

File: backend/src/Civix/CoreBundle/Entity/Group.php (hunk 2 above)
Comment (habibillah) [inline backend/src/Civix/CoreBundle/Entity/Group.php]: Sure

Comment (austinpapp) [inline backend/src/Civix/CoreBundle/Entity/Group.php]: thanks.

**Comment by austinpapp** [comment]
@habibillah sorry but there seems to be a small conflict. can you rebase and check where the conflict is. thanks for your hard work! everything looks good here. once resolved will merge.

**Comment by habibillah** [comment]
> I suggest to use \Liip\FunctionalTestBundle\Test\WebTestCase instead ...

Implemented. But still using mine :)

> @habibillah what is the test WebTestCase testing?

this used to store common method need by functional test. e.g JMS serialize/deserilize.

> @habibillah sorry but there seems to be a small conflict. can you rebase and check where the conflict is. thanks for your hard work! everything looks good here. once resolved will merge. 

Done

**Pull request merged**